### ISSUE TITLE
rtorrent not downloading torrents that Transmission on the same network will

### ISSUE THREAD

**Issue by Lusiphur** [rakshasa/rtorrent#645]
So, I fresh installed rtorrent from the Ubuntu repos on a VM in my network.
I can add a torrent that downloads fine on my Macbook on the same network using Transmission, it wont download using rtorrent.

Config (should pretty much be default)
min_peers = 40
max_peers = 100
min_peers_seed = 25
max_peers_seed = 60
max_uploads = 30
directory = <some dir>
session = /home/<user>/.session
schedule = watch_directory,5,5,load_start=<watch dir>
schedule = low_diskspace,5,60,close_low_diskspace=10240M
port_range = <ports>
port_random = yes
check_hash = yes
encryption = allow_incoming,enable_retry,prefer_plaintext (I have also tried with none)
dht = auto
dht_port = <port>

I can see the port is open using a port checker and I am sure my router is properly configured.

Ports, usernames and directory names redacted ofc.

**Comment by suburbanafro** [comment]
Assuuming this is your full .rc file, I would try a specific port range (in some obscure range like 60125-64125)  to avoid issues with your isp and maybe fixing the typo on /home//.session to /home/.session

**Comment by soredake** [comment]
Same for me, but in my case it's aria2 and rtorrent, aria2c downloads fine but rtorrent is not.

**Comment by soredake** [comment]
Seems the `protocol.encryption.set` option caused this, after deleting rtorrent started to download torrent.

**Comment by chros73** [comment]
Was it a magnet link? If so and encryption was set then it's a known issue.

**Comment by soredake** [comment]
Tried the same torrent with magnet-link, magnet-link converted to real torrent (with aria2c) and magnet-link converted to real torrent with force-encryption, nothing worked.

**Comment by chros73** [comment]
I guess the actual link/file can't be shared. :)

**Comment by soredake** [comment]
@chros73 
[magnet+torrent.zip](https://github.com/rakshasa/rtorrent/files/1475715/magnet.torrent.zip)
Torrent created with `aria2c --bt-force-encryption=true -d $PWD --bt-metadata-only=true --bt-save-metadata=true --listen-port=6881 <magnet>`

**Comment by chros73** [comment]
No worries.

It works for me (using [rtorrent-ps-ch](https://github.com/chros73/rtorrent-ps-ch), but it should also work with 0.9.6): 
- I only tried out the magnet link from the txt file with the help of [this script](https://github.com/chros73/rtorrent-ps-ch_setup/blob/master/ubuntu-14.04/home/chros73/bin/addMagnetLink.sh) that creates a fake torrent file from the given magnet link (but it should work from the load prompt as well).

So, I bet it's your settings that caused this.
- it only contains UDP trackers (only 2 worked out of them)
- and you can also use DHT (this worked as well)

Take a look [this part](https://github.com/chros73/rtorrent-ps-ch_setup/blob/master/ubuntu-14.04/home/chros73/.rtorrent-config.rc#L215) of the config I use: 
- how to enable magnet link support and DHT 
- also don't forget to open/portforward the required ports in your firewall

**Comment by soredake** [comment]
@chros73 dht and udp is enabled in my config, but port's is not forwarded, rtorrent is not supporting UPnP for port forwarding?

**Comment by chros73** [comment]
I don't think so. But I hate everything that I can't completely control (UPnP, IPv6, etc.) :)

> removal of that line

Which one? :)

I also added a new section [Supporting public torrents](https://github.com/rakshasa/rtorrent/wiki/Common-Tasks-in-rTorrent#supporting-public-torrents-magnet-links-udp-trackers-dht) onto Common Tasks WIKI page to be able to be easily referenced next time when someone states that magnet link doesn't work :)

**Comment by soredake** [comment]
@chros73 `protocol.encryption.set` line.

**Comment by wberrier** [comment]
Maybe this one is related to #759 ?

**Comment by rakshasa** [comment]
Fixed magnet links with no trackers embedded.
